components/core/CMakeLists.txt

```diff
@@ -105,37 +105,72 @@ else()
     # NOTE: On some Linux distributions (e.g. Ubuntu), the spdlog package only contains a dynamic
     # library. If the `find_package(spdlog)` call below fails, re-run
     # `tools/scripts/lib_install/<dist_name>/install-packages-from-source.sh` to build spdlog from
     # source.
     set(spdlog_USE_STATIC_LIBS ON)
 endif()
-find_package(spdlog 1.9.2 REQUIRED)
+find_package(spdlog ${spdlog_MIN_VERSION} REQUIRED)
 if(spdlog_FOUND)
     message(STATUS "Found spdlog ${spdlog_VERSION}")
 else()
+    message("You may want to re-run `components/core/tools/scripts/lib_install/<dist>/install-packages-from-source.sh`")
     if (CLP_USE_STATIC_LIBS)
-        message(FATAL_ERROR "Could not find static libraries for spdlog. You may want to re-run
-            `components/core/tools/scripts/lib_install/<dist>/install-packages-from-source.sh`")
+        message(FATAL_ERROR "Could not find static libraries for spdlog.")
     else()
         message(FATAL_ERROR "Could not find libraries for spdlog.")
     endif()
 endif()
 
+# Determine the lower bound version compatibility of fmt w.r.t to spdlog
+if(spdlog_VERSION VERSION_GREATER_EQUAL "1.15.0")
+    set(FMT_MIN_VERSION "11.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.12.0")
+    set(FMT_MIN_VERSION "10.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.11.0")
+    set(FMT_MIN_VERSION "9.0.0")
+elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.9.0")
+    set(FMT_MIN_VERSION "8.0.0")
+else()
+    message(FATAL_ERROR "Unsupported spdlog version (${spdlog_VERSION}). Minimum supported version is 1.9.0.")
+endif()
+
+# Find fmt with the compatible version
+# NOTE:
+# - We only try to link to the static library
+find_package(fmt ${FMT_MIN_VERSION} REQUIRED)
+if(fmt_FOUND)
+    # Determine the upper bound version compatibility of fmt w.r.t to spdlog
+    if(spdlog_VERSION VERSION_GREATER_EQUAL "1.15.0")
+        # Empty block, no upper version limit
+    elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.12.0")
+        if (fmt_VERSION VERSION_GREATER_EQUAL "11.0.0")
+            message(FATAL_ERROR "Unsupported fmt version (${fmt_VERSION}). Support version: 10.0.0 <= version < 11.0.0")
+        endif()
+    elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.11.0")
+        if (fmt_VERSION VERSION_GREATER_EQUAL "10.0.0")
+            message(FATAL_ERROR "Unsupported fmt version (${fmt_VERSION}). Support version: 9.0.0 <= version < 10.0.0")
+        endif()
+    elseif(spdlog_VERSION VERSION_GREATER_EQUAL "1.9.0")
+        if (fmt_VERSION VERSION_GREATER_EQUAL "9.0.0")
+            message(FATAL_ERROR "Unsupported fmt version (${fmt_VERSION}). Support version: 8.0.0 <= version < 9.0.0")
+        endif()
+    endif()
+    message(STATUS "Found fmt ${fmt_VERSION}")
+else()
+    message(FATAL_ERROR "Could not find static libraries for fmt")
+endif()
+
+# Print the selected versions for debugging
+message(STATUS "Using spdlog version ${spdlog_VERSION}")
+message(STATUS "Using fmt version ${fmt_VERSION}")
```

Comment:
These are redundant since we're already printing "Found Xxx" for each library above.